### PR TITLE
Add black formatter to pre-commit hooks

### PR DESCRIPTION
**Pull request by m453h**
This PR adds [black code formatter](https://black.readthedocs.io/en/stable/#) to pre-commit hooks to enforce a consistent code style across the project, resolving issue #96.  As a result, numerous files have been reformatted without altering any functionality.

### CODE REVIEW

**Comment by kilemensi** [comment]
👍🏽 ... Don't forget to ensure Black plays nice with [other tools](https://black.readthedocs.io/en/stable/guides/using_black_with_other_tools.html), at least `isort` and `flake8`

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -15,6 +15,11 @@ repos:
 #    hooks:
 #      - id: mypy
 #        entry: mypy
+  - repo: http://github.com/ambv/black
+    rev: 24.4.2
+    hooks:
+      - id: black
+        language_version: python3.10
```
Comment (kilemensi) [inline .pre-commit-config.yaml:22]: I think
```yaml
  - repo: http://github.com/pycqa/isort
```
is missing, no?

**Review by kilemensi — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (m453h) [reply .pre-commit-config.yaml:22]: No this is not missing

File: pyproject.toml [hunk 2]
```diff
@@ -42,11 +42,14 @@ name = "mcmetadata"
 
 [project.optional-dependencies]
 dev = [
-    "pre-commit", "flake8", "mypy", "isort", "types-urllib3", "types-requests"
+    "pre-commit", "flake8", "mypy", "isort", "types-urllib3", "types-requests", "black"
```
[the remark this answers is not in the document]
Comment (m453h) [reply pyproject.toml:45]: Hmm, I could be wrong but looking at the usage of the dependencies wouldn't what is specified in `pre-commit-config.yaml` be enough ? I'm just wondering which would be the source of truth for the versions🤔

[the remark this answers is not in the document]
Comment (kilemensi) [reply pyproject.toml:45]: `precommit` creates its own `env` when run, `pip` (or any other dependency management tool) won't know anything about what's defined in `pre-commit-config.yaml`

File: .pre-commit-config.yaml (hunk 1 above)
Comment (kilemensi) [reply .pre-commit-config.yaml:22]: Not sure how I missed it but yep, it's there.

File: pyproject.toml (hunk 2 above)
[the remark this answers is not in the document]
Comment (kilemensi) [reply pyproject.toml:45]: On source of truth, ideally, `pyproject.toml` should be the source of truth for all tools, but this is python ecosystem we're talking about.

[the remark this answers is not in the document]
Comment (m453h) [reply pyproject.toml]: I have pinned the specific versions, I believe we should be good.

**Pull request merged**